### PR TITLE
add command to run prisma format

### PR DESCRIPTION
**Pull request by nverno**
This adds a simple interactive command to run `prisma format` on the buffer's file to `prisma-mode-map`.  Also, adds basic imenu expression for models/enums.

### CODE REVIEW

**Comment by pimeys** [comment]
Cool thanks. So far I've been just using `lsp-format-buffer` which uses the `prisma-language-server` to format the buffer.

File: prisma-mode.el [hunk 1]
```diff
@@ -67,6 +67,23 @@
...
           )))
 
+(defun prisma-format-buffer ()
+  "Run prisma format on current prisma buffer.
+Attempts to find prisma by searching up directory tree for package.json and
+associated node_modules/.bin/prisma."
+  (interactive)
+  (let* ((buf (buffer-file-name))
+         (pkg (and buf (locate-dominating-file buf "package.json")))
+         (prisma (and pkg (expand-file-name "node_modules/.bin/prisma" pkg))))
```
Comment (pimeys) [inline prisma-mode.el:77]: This will break with nixos.

**Comment by pimeys** [comment]
So, basically the path to prisma binary should not be absolute. For example I'm using nixos, and prisma is in a different directory depending on your OS version. Could it be enough for you to use `lsp-mode` together with `prisma-language-server` to get what you need?

**Comment by nverno** [comment]
yea, for sure, if it's not helpful just close it.  Do you want imenu support? I just realized I should also set the `imenu-create-index-function` to the default if it is defined as `js--imenu-create-index`

**Comment by pimeys** [comment]
Everything you think can be useful for you, but it must be compatible with different systems. No absolute paths to binaries, and it's quite easy for me to merge.

**Comment by nverno** [comment]
I forgot about `lsp-format-buffer`, that does seem like the way to go for that, thanks!

**Comment by nverno** [comment]
OK, Ill close this and make a new one with just the imenu stuff - it is compatible and just a couple lines of code- do you use lsp for imenu?

**Pull request closed without merging**